### PR TITLE
CENG-1661: Upgrade to 3.3

### PR DESCRIPTION
**Pull request by peterjmag**
Pulls in changes from the latest upstream master, including the [3.3 release](https://github.com/facebook/create-react-app/releases/tag/v3.3.0).

Note that 3.3 supports [custom templates](https://create-react-app.dev/docs/custom-templates/), but I'm taking advantage of that yet for this PR — still just overriding the default templates.

Here's the list of commits from the upstream repo:
https://github.com/digitalocean/create-react-app/compare/ce8bc157ac0822102c3704e922bdfffe7b1ad9f8...facebook:88a543549bd3cfa0dcc77d3abd8533a3c1ff9bca

And here are the merge conflicts I resolved as part of this:

```
Unmerged paths:
  (use "git add <file>..." to mark resolution)

	both modified:   packages/cra-template-typescript/template/public/index.html
	added by them:   packages/cra-template-typescript/template/src/App.css
	added by them:   packages/cra-template-typescript/template/src/index.css
	added by them:   packages/cra-template-typescript/template/src/serviceWorker.ts
	both modified:   packages/cra-template/template/public/index.html
	added by them:   packages/cra-template/template/src/App.css
	added by them:   packages/cra-template/template/src/serviceWorker.js
	both modified:   packages/react-scripts/package.json
	both modified:   packages/react-scripts/scripts/init.js
```

### CODE REVIEW

File: packages/react-scripts/scripts/init.js [hunk 1]
```diff
@@ -80,40 +80,74 @@ module.exports = function(
...
+
+  let templateJson = {};
+  if (fs.existsSync(templateJsonPath)) {
+    templateJson = require(templateJsonPath);
+  }
+
   // Copy over some of the devDependencies
   appPackage.dependencies = appPackage.dependencies || {};
 
-  // Add walrus
```
Comment (philtobias) [inline packages/react-scripts/scripts/init.js]: Was this moved elsewhere or accidentally deleted?

File: packages/react-scripts/scripts/init.js [hunk 2]
```diff
@@ -80,40 +80,74 @@ module.exports = function(
...
+      'test:ci': 'CI=true npm-run-all stylelint format:check test',
+    },
+    templateScripts
+  );
 
-  if (useTypeScript) {
-    appPackage.scripts['test:ci'] =
-      'CI=true npm-run-all tslint stylelint format:check test';
-    appPackage.scripts.tslint = 'tslint --project tsconfig.json';
-    appPackage.scripts['tslint:fix'] = 'tslint --project tsconfig.json --fix';
```
Comment (philtobias) [inline packages/react-scripts/scripts/init.js]: Same here - was this moved or accidentally deleted?

File: packages/react-scripts/scripts/init.js (hunk 1 above)
Comment (peterjmag) [reply packages/react-scripts/scripts/init.js]: Good catch, I'll fix up this one and the other one!

Comment (peterjmag) [reply packages/react-scripts/scripts/init.js]: Addressed in https://github.com/digitalocean/create-react-app/pull/11/commits/bc4ba5ae09a41c51f7b07ee60485de060491d9a1. I also updated Walrus to the latest version in a separate commit.

File: packages/react-scripts/scripts/init.js (hunk 2 above)
Comment (peterjmag) [reply packages/react-scripts/scripts/init.js]: Addressed in https://github.com/digitalocean/create-react-app/pull/11/commits/bc4ba5ae09a41c51f7b07ee60485de060491d9a1.

**Pull request merged**